src/pybind/mgr/cephadm/serve.py

```diff
@@ -1598,6 +1585,39 @@ async def _get_container_image_info(self, image_name: str) -> ContainerInspectIn
         self.log.debug(f'image {image_name} -> {r}')
         return r
 
+    def _set_cephadm_command_timeout(self, timeout: Any) -> int:
+        def _safe_convert_to_int(t: Any) -> int:
+            if not isinstance(t, int):
+                try:
+                    t = int(t)
+                except Exception:
+                    # just use a default if we can't convert to an int
+                    self.log.info(f'Got cephadm command timeout {t} of type '
+                                  f'{type(t)} that could not be converted to '
+                                  'an integer. Using default 900')
+                    t = 900
+            return t
```

Comment:
I like the idea of enforcing the type but I'm not sure it's the right place do do it. Reviewing the code I cannot see how we end up with a float. I think we should address the issue there instead of trying to "cast" the type later in the code. WDYT?

Reply:
The issue is I'm not sure exactly how it became a float either. I don't think it's anything cephadm was doing with the value (we only read it, writing it is done entirely through setting the config option), so it's likely something to do with the general mgr module handling of the values. I didn't want to get into that, and thought making sure it was the correct type as close as possible to when we really need it to be made sense.